reactful-numbers/src/app/components/ButtonsBox.js

```diff
@@ -1,67 +1,168 @@
 import React from "react";
 import Button from "./Button";
-function ButtonsBox() {
+
+function ButtonsBox({
+  setInputValue,
+  clearInput,
+  invertSign,
+  percentage,
+  expressionBoxValue,
+  equals,
+}) {
   return (
     <div className="buttons-box">
       <div className="grid-container">
         <div>
-          <Button buttonColor="grey" buttonText="AC"></Button>
+          <Button
+            buttonColor="grey"
+            buttonText={expressionBoxValue == "" ? "AC" : "C"}
+            onClick={() => clearInput()}
+          ></Button>
         </div>
         <div>
-          <Button buttonColor="grey" buttonText="+/-"></Button>
+          <Button
+            buttonColor="grey"
+            buttonText="+/-"
+            onClick={() => invertSign()}
+          ></Button>
         </div>
         <div>
-          <Button buttonColor="grey" buttonText="%"></Button>
+          <Button
+            buttonColor="grey"
+            buttonText="%"
+            onClick={() => percentage()}
+          ></Button>
         </div>
         <div>
-          <Button buttonColor="orange" buttonText="/"></Button>
+          <Button
+            buttonColor="orange"
+            buttonText="/"
+            onClick={
+              String(expressionBoxValue).includes("/")
+                ? () => equals()
+                : () => setInputValue("/")
+            }
+          ></Button>
         </div>
         <div>
-          <Button buttonColor="dark-grey" buttonText="7"></Button>
+          <Button
+            buttonColor="dark-grey"
+            buttonText="7"
+            onClick={() => setInputValue("7")}
+          ></Button>
         </div>
         <div>
-          <Button buttonColor="dark-grey" buttonText="8"></Button>
+          <Button
+            buttonColor="dark-grey"
+            buttonText="8"
+            onClick={() => setInputValue("8")}
+          ></Button>
         </div>
         <div>
-          <Button buttonColor="dark-grey" buttonText="9"></Button>
+          <Button
+            buttonColor="dark-grey"
+            buttonText="9"
+            onClick={() => setInputValue("9")}
+          ></Button>
         </div>
         <div>
-          <Button buttonColor="orange" buttonText="%"></Button>
+          <Button
+            buttonColor="orange"
+            buttonText="*"
+            onClick={
+              String(expressionBoxValue).includes("*")
```

Comment:
This along with all other operators implements similar functionality to the IOS calculator where every time you insert a second operator in the expression box value, the previous expression get evaluelated